solvers/modopt_fista.py

```diff
@@ -0,0 +1,107 @@
+import numpy as np
+
+from benchopt import BaseSolver
+from benchopt import safe_import_context
+
+
+with safe_import_context() as import_ctx:
+    from modopt.opt.algorithms import ForwardBackward
+    from modopt.opt.proximity import SparseThreshold
+    from modopt.opt.linear import Identity
+    from modopt.opt.gradient import GradBasic
+
+
+class Solver(BaseSolver):
+    name = 'ModOpt-FISTA'
+    stop_strategy = 'iteration'
+    install_cmd = 'conda'
+    requirements = [
+        'pip:git+https://github.com/CEA-COSMIC/ModOpt.git',
+    ]
+    parameters = {
+        'restart_strategy': ['greedy', 'adaptive-1'],
+    }
+    references = [
+        'S. Farrens, A. Grigis, L. El Gueddari, Z. Ramzi, G. R. Chaithya, '
+        'S. Starck, B. Sarthou, H. Cherkaoui, P. Ciuciu and J.-L. Starck, '
+        '"PySAP: Python Sparse Data Analysis Package for multidisciplinary '
+        'image processing", Astronomy and Computing, vol. 32, '
+        ' pp. 100402 (2020)'
+    ]
+    support_sparse = False
+
+    def skip(self, X, y, lmbd, fit_intercept):
+        if fit_intercept:
+            return True, f"{self.name} does not handle fit_intercept"
+
+        return False, None
+
+    def set_objective(self, X, y, lmbd, fit_intercept):
+        self.X, self.y, self.lmbd = X, y, lmbd
+        self.fit_intercept = fit_intercept
+
+        n_samples, n_features = self.X.shape
+        x_shape = n_features
+        if fit_intercept:
+            x_shape += n_samples
+        self.var_init = np.zeros(x_shape)
+        if self.restart_strategy == 'greedy':
+            min_beta = 1.0
```

Comment:
it seems that `min_beta` is not set afterwards, but it should rather be 1/L, this may explain the divergence 

Also @zaccharieramzi said setting parameters after init may not be a good idea, so instead we could compute L here, set the correct value in init, then recompute L in `run()` for fairness

Reply:
that sounds fair.
Also thanks for the catch @mathurinm